File: back-end/src/main/java/com/huyphan/dtos/CommentDto.java
package com.huyphan.dtos;

import lombok.Getter;
import lombok.Setter;
import org.springframework.lang.Nullable;

@Getter
@Setter
public class CommentDto {

    private Long id;

    @Nullable
    private String text;

    @Nullable
    private String mediaUrl;

    private int upvotes;

    private int downvotes;

    @Nullable
    private String mediaType;

    @Nullable
    private UserDto replyTo;

    private String date;

    @Nullable
    private UserDto user;

    private Long totalChildren;

    private boolean isUpvoted;

    private boolean isDownvoted;
    private boolean notificationEnabled;
    private Long parentId;

    private boolean followed;

    private Long postId;

    private Long replyToId;

    private boolean nsfw;

    public boolean getIsDownvoted() {
        return isDownvoted;
    }

    public boolean getIsUpvoted() {
        return isUpvoted;
    }
}

File: client/src/services/dtos/comment-dto.ts
import { UserDto } from './user-dto';

export default interface CommentDto {
  readonly id: number;

  readonly text: string;

  readonly mediaUrl: string | null;

  readonly upvotes: number;

  readonly downvotes: number;

  readonly mediaType: string | null;

  readonly replyTo: UserDto | null;

  readonly date: string;

  readonly user: UserDto;

  readonly totalChildren: number;

  readonly isUpvoted: boolean;

  readonly isDownvoted: boolean;

  readonly parentId: number | null;

  readonly postId: number;

  readonly replyToId: number | null;

  readonly notificationEnabled: boolean;

  readonly followed: boolean;

  readonly nsfw: boolean;

  readonly moderating: boolean;
}
